### DISCUSSION TITLE
Integration tests for `sea-orm-cli` and `codegen`?

### DISCUSSION THREAD

**Discussion by JonasCir** [SeaQL/sea-orm#1363] [Q&A]
Hey there,
I could contribute a small testing framework for `sea-orm-cli` which would invoke the CLI and then asserts that the generated code in the filesystem matches an expected folder structure/content. I did contributed similar logic to other projects already, the most recent example can be found [here](https://github.com/EpicsDAO/zapp/blob/main/tests/mod.rs). Is this something you are interested in? It usually gives me a good feeling to have tests to ensure that the complete generation works.

Besides this, I'm a little bit confused how `sea-orm-codegen` is tested. I see an integration test [folder](https://github.com/SeaQL/sea-orm/tree/master/sea-orm-codegen/tests), however, I cannot find any `#[test]` annotated functions if I'm not mistaken?

**Comment by billy1624** [comment]
Hey @JonasCir, thanks for the initiatives!

A testing framework for `sea-orm-cli` would be awesome to have! Feel free to file a PR!

Regarding the `sea-orm-codegen`, the tests is defined at the writer

https://github.com/SeaQL/sea-orm/blob/ecb299a936b5e65654e72b46af4065cc30950b4f/sea-orm-codegen/src/entity/writer.rs#L779-L1958

**Reply by JonasCir** [reply]
Ok, great! I put it on my list :)
